tardis/conftest.py

```diff
@@ -35,6 +42,54 @@ def pytest_addoption(parser):
     parser.addoption("--slow-test-data", dest="slow-test-data",
                      help="path to directory having baseline data for slow tests")
 
+    # Accept dokuwiki credentials through command line.
+    parser.addoption("--username", dest='username', help="Dokuwiki username")
+    parser.addoption("--password", dest='password',help="Dokuwiki password")
+
+
+def pytest_configure(config):
+    html_file = tempfile.NamedTemporaryFile(delete=False)
+    # Html test report will be generated at this filepath by pytest-html plugin
+    config.option.htmlpath = html_file.name
+
+
+def pytest_unconfigure(config):
+    # Html report created by pytest-html plugin is read here, uploaded to
+    # dokuwiki and finally deleted.
+    try:
```

Comment:
you should do a try and except statement at the beginning with then just having dokuwiki_available boolean variable.